pandas/tests/series/test_dtypes.py

```diff
@@ -153,6 +153,38 @@ def test_astype_dict(self):
         self.assertRaises(KeyError, s.astype, {'abc': str, 'def': str})
         self.assertRaises(KeyError, s.astype, {0: str})
 
+    def test_astype_generic_timestamp(self):
+        # see gh-15524
+        data = [1]
+
+        s = Series(data)
+        dtype = np.datetime64
+        result = s.astype(dtype)
+        expected = Series(data, dtype=dtype)
+        assert_series_equal(result, expected)
+
+        s = Series(data)
+        dtype = np.timedelta64
+        result = s.astype(dtype)
+        expected = Series(data, dtype=dtype)
+        assert_series_equal(result, expected)
+
+    def test_astype_empty_constructor_equality(self):
+        # see gh-15524
+
+        for dtype in np.typecodes['All']:
```

Comment:
There actually is no example where you create a class that inherits from `object`.  However, that being said, having been able to sit down and re-examine the `class` suggestion, I do see what you mean now.

Several things can come from this discussion:

1) Document how you can instantiate test classes that inherit from `object` (and not `tm.TestCase`).  I still prefer writing test classes instead of raw functions for organizational reasons, though it will still follow `pytest` idiom.

2) Make a new PR that modifies the changes in #15952 to use `pytest` fixtures for redirecting `stdout` and `stdin`.

3) Modify the existing test class in this file to inherit only from `object` (new commit).